### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by serhido**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,8 +1,30 @@
...
+
+    def calculate_washing_price(self, car: Car):
+        price = (car.comfort_class * (self.clean_power - car.clean_mark) * self.average_rating / self.distance_from_city_center)
+        return round(price)
+
+    def wash_single_car(self, car: Car):
+        car.clean_mark = self.clean_power
+
+    def rate_service(self, rate: int):
+        return round((self.average_rating * self.count_of_ratings + rate) / (self.count_of_ratings + 1), 1)
```
Comment (ArsenPidhoretskyi) [inline app/main.py:30]: No new line exist

File: app/main.py [hunk 2]
```diff
@@ -1,8 +1,30 @@
...
 class CarWashStation:
-    # write your code here
-    pass
+    def __init__(self, distance_from_city_center: float, clean_power: int, average_rating: float, count_of_ratings: int) -> None:
+        self.distance_from_city_center = distance_from_city_center
+        self.clean_power = clean_power
+        self.average_rating = round(average_rating, 1)
+        self.count_of_ratings = count_of_ratings
+
+    def serve_cars(self, list_cars: list):
```
Comment (ArsenPidhoretskyi) [inline app/main.py:14]: list_cars is not okay name
Use cars
Also list of what? Add type hint

**Review by ArsenPidhoretskyi — CHANGES_REQUESTED** [review]
Use flake8 lines are to long

File: app/main.py [hunk 3]
```diff
@@ -1,8 +1,35 @@
...
+    def serve_cars(self, cars: list) -> int:
+        result = 0
+        for car in cars:
+            if car.clean_mark < self.clean_power:
+                self.wash_single_car(car)
+                result += self.calculate_washing_price(car)
+        return result
+
+    def calculate_washing_price(self, car: Car) -> int:
+        price = (car.comfort_class * (self.clean_power - car.clean_mark) * self.average_rating / self.distance_from_city_center)
```
Comment (vasylhnatiuk) [inline app/main.py:28]: Use flake8, lines are to long in several places

**Review by vasylhnatiuk — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 4]
```diff
@@ -1,8 +1,35 @@
...
+        return round(price)
+
+    def wash_single_car(self, car: Car) -> None:
+        car.clean_mark = self.clean_power
+
+    def rate_service(self, rate: int) -> float:
+        return round((self.average_rating * self.count_of_ratings + rate) / (self.count_of_ratings + 1), 1)
```
Comment (vasylhnatiuk) [inline app/main.py:35]: the same here

**Review by vasylhnatiuk — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 5]
```diff
@@ -1,8 +1,45 @@
...
+
+    def wash_single_car(self, car: Car) -> None:
+        car.clean_mark = self.clean_power
+
+    def rate_service(self, rate: int) -> float:
+        self.average_rating = round(
+            (self.average_rating * self.count_of_ratings + rate)
+            / (self.count_of_ratings + 1), 1)
+        self.count_of_ratings += 1
+        return self.average_rating
```
Comment (vasylhnatiuk) [inline app/main.py:45]: return is useless here

File: app/main.py [hunk 6]
```diff
@@ -1,8 +1,45 @@
...
-    pass
+    def __init__(self,
+                 distance_from_city_center: float,
+                 clean_power: int,
+                 average_rating: float,
+                 count_of_ratings: int
+                 ) -> None:
+        self.distance_from_city_center = distance_from_city_center
+        self.clean_power = clean_power
+        self.average_rating = round(average_rating, 1)
```
Comment (vasylhnatiuk) [inline app/main.py:21]: round is useless here

File: app/main.py [hunk 7]
```diff
@@ -1,8 +1,45 @@
 class Car:
```
Comment (vasylhnatiuk) [inline app/main.py:1]: =========================== short test summary info ============================
FAILED tests/test_main.py::test_car_wash_station[cars1-wash_station1-22.4] - ...
FAILED tests/test_main.py::test_car_wash_station[cars3-wash_station3-40.3] - ...
FAILED tests/test_main.py::test_car_wash_station[cars4-wash_station4-13.0] - ...
========================= 3 failed, 17 passed in 0.16s =========================

you fail 3 test

**Review by vasylhnatiuk — CHANGES_REQUESTED** [review]

**Review by Grayder0152 — APPROVED** [review]

**Pull request closed without merging**